src/type.coffee

```diff
@@ -48,7 +48,7 @@ module.exports = class Type extends Emitter
 
   _getByIDs: (ids, otherArgs...) ->
     requests = for id in ids
-      if id of @_resourcesCache and otherArgs.length is 0
+      if id of @_resourcesCache and (otherArgs.length is 0 or not otherArgs[0]?)
```

Comment:
The variable query [here](https://github.com/edpaget/json-api-client/blob/load-linked-resources-from-cache/src/resource.coffee#L133) can be undefined if you're not passing any query params with the link fetch. If you pass any argument even undefined this function skips the cache.